desktop/extensions-sdk/guides/invoke-host-binaries.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Invoke host binaries
+description: Add invocations to host binaries from the frontend with the extension SDK.
+keywords: Docker, extensions, sdk, build
+---
+
+In some cases, your extension needs to invoke some command from the host (the computer of your users). For example, you 
+might wand to invoke the CLI of your cloud provider to create a new resource, or the CLI of a tool your extension 
+provides. 
+You could do that executing the CLI from a container with the extension SDK. But this CLI needs to access the host's 
+filesystem, which isn't possible if it runs in a container.
+Host binaries allows exactly this: to invoke binaries from the host. As extensions can run on multiple platforms, 
+this means that you need to ship the binaries for all the platforms you want to support.
+
+In this example, this CLI will be a simple `Hello world` script that must be invoked with a parameter and will return a 
+string. Since the extension will support multiple platforms, the script will be written in `bash` for macOS and Linux, 
+and in `powershell` for Windows.
+
+## Add the binaries to the extension
+Create a file `binaries/unix/hello.sh` with the following content:
+
+```bash
+#!/bin/sh
+echo "Hello, $1!"
+```
+
+Create another file `binaries/windows/hello.ps1` with the following content:
+
+```powershell
+Write-Output "Hello, $args[0]!"
+```
+
+Make them both executable:
+
+```bash
+chmod +x binaries/*
```

Comment:
If I'm following this tutorial on Windows, `chmod` won't work

Reply:
Indeed, I need to figure out if it is needed and how to make it work. I haven't tested it yet on windows. I wanted the team's feedback before pursuing it. Thanks for raising this though. Good reminder!

Reply:
Indeed, I need to figure out if it is needed and how to make it work. I haven't tested it yet on windows. I wanted the team's feedback before pursuing it. Thanks for raising this though. Good reminder!